File: hyerang0125/0x07/boj_10866.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    cin.tie(0); ios::sync_with_stdio(0);

    int x, n;
    string str;
    deque<int> dq;

    cin >> n;
    while(n--){
        cin >> str;
        if(str == "push_front"){
            cin >> x;
            dq.push_front(x);
        }
        else if(str == "push_back"){
            cin >> x;
            dq.push_back(x);
        }
        else if(str == "pop_front"){
            if(dq.empty()) cout << "-1\n";
            else{
                cout << dq.front() << "\n";
                dq.pop_front();
            }
        }
        else if(str == "pop_back"){
            if(dq.empty()) cout << "-1\n";
            else{
                cout << dq.back() << "\n";
                dq.pop_back();
            }
        }
        else if(str == "size"){
            cout << dq.size() << "\n";
        }
        else if(str == "empty"){
            if(dq.empty()) cout << "1\n";
            else cout << "0\n";
        }
        else if(str == "front"){
            if(dq.empty()) cout << "-1\n";
            else cout << dq.front() << "\n";
        }
        else if(str == "back"){
            if(dq.empty()) cout << "-1\n";
            else cout << dq.back() << "\n";
        }
    }

    return 0;
}

File: minyoung/BOJ_10866.java
import java.util.Deque;
import java.util.LinkedList;
import java.util.StringTokenizer;
import java.io.BufferedReader;
import java.io.InputStreamReader;


public class Main
{
	public static int n;
	
	public static void main (String[] args) throws java.lang.Exception
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();		
		StringTokenizer st;
		
		Deque <Integer> deque = new LinkedList<>();
		
		
		n = Integer.parseInt(br.readLine());
			
		int result=0;
		for(int i=0;i<n;i++) {
			
			st = new StringTokenizer(br.readLine());
			
			String str = st.nextToken();
			
			switch(str) {
				case "push_back":
					deque.offerLast(Integer.parseInt(st.nextToken()));
					break;
				case "push_front":
					deque.offerFirst(Integer.parseInt(st.nextToken()));
					break;
				case "pop_front":
					result = deque.isEmpty()? -1 : deque.pollFirst();
					sb.append(result).append("\n");
					break;
				case "pop_back":
					result = deque.isEmpty()? -1 : deque.pollLast();
					sb.append(result).append("\n");
					break;
				case "size":
					sb.append(deque.size()).append("\n");
					break;
				case "empty":
					result = deque.isEmpty()? 1 : 0;
					sb.append(result).append("\n");
					break;
				case "front":
					result = deque.isEmpty()? -1 : deque.getFirst();
					sb.append(result).append("\n");
					break;
				case "back":
					result = deque.isEmpty()? -1 : deque.getLast();
					sb.append(result).append("\n");
					break;
			}
			
		
		}
		
		System.out.println(sb);
	}
}
